### PR TITLE
AIK-3154

### PR DESCRIPTION
**Pull request by bitterpanda63**
Can't login to Google atm to copy Jira title, quick recap : This PR contains the following :
- Wrap 'n hook for both flask and django
- Middleware for both flask and django (django in seperate file for import)
- Reconfig of sample-apps that also allows to directly install the "aikido_firewall" module
- Switch to poetry.lock and build config

### CODE REVIEW

File: aikido_firewall/middleware/django.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import logging
+class AikidoMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request, *args, **kwargs):
+        logging.critical("[AIK] Aikido middleware : call")
```
Comment (willem-delbare) [inline aikido_firewall/middleware/django.py:7]: needs critical?

File: aikido_firewall/sources/django.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import importhook
+import copy
+from importlib.metadata import version
+
+AIKIDO_MIDDLEWARE_ADDR = "aikido_firewall.middleware.django.AikidoMiddleware"
+
+
+@importhook.on_import('django.conf')
```
Comment (willem-delbare) [inline aikido_firewall/sources/django.py:9]: i'd comment here what we are hooking and what the goal is

File: aikido_firewall/middleware/django.py (hunk 1 above)
Comment (bitterpanda63) [reply aikido_firewall/middleware/django.py:7]: Yeah.. I tested it with .info() and did not work, both flask and django suppress logs and prolly store them somewhere else, maybe we could look into the config options for our apps to enable all logs. Or use another logging system, but that should best be a new PR imo

File: aikido_firewall/sources/django.py (hunk 2 above)
Comment (bitterpanda63) [reply aikido_firewall/sources/django.py:9]: Done

**Pull request merged**